src/main/webapp/app/entities/gene/gene-curate.tsx

```diff
@@ -108,76 +95,74 @@ export const GeneCurate = (props: IGeneCurateProps) => {
+                  onRefseqChange={e => {
+                    props.updateReviewableContent(firebaseParentPath, 'dmp_refseq_id_grch38', e.target.value);
                   }}
                 />
               </div>
-              <DefaultField
+              <RealtimeBasicInput
                 label="Summary"
                 labelClass="font-weight-bold"
                 name="geneSummary"
-                type="textarea"
+                type={RealtimeInputType.TEXTAREA}
                 value={props.data?.summary || ''}
-                onChange={event => {
-                  props.update(firebaseParentPath, { summary: event.target.value });
+                onChange={e => {
+                  props.updateReviewableContent(firebaseParentPath, 'summary', e.target.value);
                 }}
               />
               <div className="mb-2 d-flex">
-                <DefaultField
-                  name="type"
-                  type="checkbox"
+                <RealtimeBasicInput
                   label="Tumor Supressor"
+                  name="tumorSupressor"
+                  type={RealtimeInputType.CHECKBOX}
                   inputClass="ml-1 position-relative"
                   checked={!!props.data?.type?.tsg}
                   onChange={e => {
-                    handleTumorSuppressorChange(e.target.checked);
+                    props.updateGeneType(firebaseParentPath, TUMOR_SUPPRESSOR, e.target.checked);
                   }}
                 />
-                <DefaultField
+                <RealtimeBasicInput
                   className="ml-4"
-                  name="type"
-                  type="checkbox"
                   label="Oncogene"
+                  name="oncogene"
+                  type={RealtimeInputType.CHECKBOX}
                   inputClass="ml-1 position-relative"
                   checked={!!props.data?.type?.ocg}
                   onChange={e => {
-                    handleOncogeneChange(e.target.checked);
+                    props.updateGeneType(firebaseParentPath, ONCOGENE, e.target.checked);
                   }}
                 />
               </div>
             </Col>
           </Row>
           <Row className={getSectionClassName()}>
             <Col>
-              <DefaultField
-                className="mb-1"
+              <RealtimeBasicInput
                 label="Background"
                 labelClass="font-weight-bold"
                 name="geneBackground"
-                type="textarea"
+                type={RealtimeInputType.TEXTAREA}
                 value={props.data?.background || ''}
-                rows={Math.ceil(props.data?.background.length / 250)}
-                onChange={event => {
-                  props.update(firebaseParentPath, { background: event.target.value });
+                onChange={e => {
+                  props.updateReviewableContent(firebaseParentPath, 'background', e.target.value);
```

Comment:
We should have the path written in constants too. Even better if we could have an util to generate the path.